homeassistant/components/modbus/cover.py

```diff
@@ -105,7 +109,13 @@ async def async_added_to_hass(self):
         """Handle entity which will be added."""
         state = await self.async_get_last_state()
         if state:
-            self._value = state.state
+            convert = {
+                STATE_CLOSED: self._state_closed,
+                STATE_CLOSING: self._state_closing,
+                STATE_OPENING: self._state_opening,
+                STATE_OPEN: self._state_open,
+            }
+            self._value = convert[state.state]
```

Comment:
Is it impossible for the state to be unknown or unavailable at this point?

Reply:
Unavailable can in theory happen, but I cannot see unknown happen.

However one thing is theory, Murphy learns us something different, so I have a added the 2 states, thanks for catching it.